daffodil-core/src/main/scala/org/apache/daffodil/dpath/NodeInfoUtils.scala

```diff
@@ -86,7 +86,7 @@ object NodeInfoUtils {
       case Decimal => Decimal
       case Integer => Decimal
       case Double => Double
-      case Long => Double
+      case Long => Decimal
       case Float => Float
       case Int => Float
       case ArrayIndex => ArrayIndex
```

Comment:
Should Int also convert to Decimal isntead of Float? So only float and double divisions return float/double, eveything else will return Decimal?

Reply:
I believe you are correct, according to the xpath spec, 'div' on integers should also return a decimal

Reply:
I'll make the change to the unsigned variants as well.

Reply:
Looking at this divResult function, it is lacking many of the smaller types (byte, short, etc) as well as all the unsigned types.  Are these types actually not supported or is this an oversight?

Reply:
The parameter passed into the divResult method comes from the second value of tuple in the match case below this. That tuple value is always one of Decimal, Double, Float, Integer, ArrayIndex, Long, or Int. So I think this match/case is exhastive. Whehter or not that's correct, I'm not 100% sure.